tests/api_connexion/endpoints/test_dag_endpoint.py

```diff
@@ -144,6 +144,36 @@ def _create_dag_model_for_details_endpoint(self, dag_id, session=None):
         )
         session.add(dag_model)
 
+    @provide_session
+    def _create_dag_model_for_details_endpoint_with_dataset_expression(self, dag_id, session=None):
+        dag_model = DagModel(
+            dag_id=dag_id,
+            fileloc="/tmp/dag.py",
+            schedule_interval="2 2 * * *",
+            is_active=True,
+            is_paused=False,
+            dataset_expression={
+                "__type": "dataset_any",
```

Comment:
I wonder if we should clean this up. The REST API is public (not just communication between backend and frontend) so we might need to better decouple this to internals.

Reply:
+1 to cleaning up the returned object. What happened to a format like?
```
"dataset_expression": {
    "any": [
      "s3://dag1/output_1.txt",
      {
        "all": [
          "s3://dag2/output_1.txt",
          "s3://dag3/output_3.txt"
        ]
      }
    ]
  },
  ```

Reply:
Sure I can do that. I just exposed it as it is

Reply:
Changed it